### PR TITLE
Windowed value coder

### PR DESCRIPTION
**Pull request by robertwb**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by robertwb** [comment]
R: @aaltay

File: sdks/python/apache_beam/coders/coder_impl.py [hunk 1]
```diff
@@ -326,8 +319,8 @@ def estimate_size(self, unused_value, nested=False):
 
 class TimestampCoderImpl(StreamCoderImpl):
 
-  def __init__(self, timestamp_class):
-    self.timestamp_class = timestamp_class
+  def __init__(self, timestamp_class=None):
+    self.timestamp_class = Timestamp#timestamp_class
```
Comment (aaltay) [inline sdks/python/apache_beam/coders/coder_impl.py]: Left over comment `#timestamp_class` ?

Actually is there a need for the `timestamp_class` argument any more?

File: sdks/python/apache_beam/transforms/timeutil.py [hunk 2]
```diff
@@ -25,196 +25,11 @@
...
-    return Duration(micros=self.micros * other.micros / 1000000)
-
-  def __rmul__(self, other):
-    return self * other
-
-  def __mod__(self, other):
-    other = Duration.of(other)
-    return Duration(micros=self.micros % other.micros)
-
+# For backwards compatibility
```
Comment (aaltay) [inline sdks/python/apache_beam/transforms/timeutil.py]: Should we mark them as deprecated? Would it make sense to remove them later?

File: sdks/python/apache_beam/coders/coder_impl.py (hunk 1 above)
Comment (robertwb) [inline sdks/python/apache_beam/coders/coder_impl.py]: Nope. Gone.

File: sdks/python/apache_beam/transforms/timeutil.py (hunk 2 above)
Comment (robertwb) [inline sdks/python/apache_beam/transforms/timeutil.py]: Can't programmatically deprecate this, but I added a TODO.

**Pull request closed without merging**